File: Caesar/caesar.cpp
#include <iostream>
using namespace std;

void encrypt(string plain, int shift) {
    string cipher = "";
    int i;

    for (i = 0; i < plain.length(); i++) {
        // cout << "is here";
        if (isspace(plain[i])) {
            cipher[i] = plain[i];
        } else {
            if (isupper(plain[i])) {
                cipher[i] += char(int(plain[i] + shift - 65) % 26 + 65);
            } else {
                cipher[i] += char(int(plain[i] + shift - 97) % 26 + 97);
            }
        }
        cout << cipher[i];
    }
}

void decrypt(string cipher, int shift) {
    string plain = "";
    int i;

    for (i = 0; i < cipher.length(); i++) {
        if (isspace(cipher[i])) {
            plain[i] = cipher[i];
        } else {
            if (isupper(cipher[i])) {
                plain[i] += char(int(cipher[i] - shift - 65) % 26 + 65);
            } else {
                plain[i] += char(int(cipher[i] - shift - 97) % 26 + 97);
            }
        }
        cout << plain[i];
    }
}

int main() {
    int choice, shift;
    string text;
    cout << "Encrypt (1) or Decrypt (2): ";
    cin >> choice;

    if (choice == 1) {
        cout << "Enter your text: ";
        cin >> text;
        cout << "Enter your shift: ";
        cin >> shift;
        cout << "Plaintext: " << text;
        cout << "\nCiphertext: ";
        encrypt(text, shift);
    } else {
        cout << "Enter your text: ";
        cin >> text;
        cout << "Enter your shift: ";
        cin >> shift;
        cout << "Ciphertext: " << text;
        cout << "\nPlaintext: ";
        decrypt(text, shift);
    }

    return 0;
}

File: Caesar/caesar.py
def encrypt(text, s):
    result = ""
 
    for i in range(len(text)):
        char = text[i]
        if (char.isupper()):
            result += chr((ord(char) + s - 65) % 26 + 65)
        else:
            result += chr((ord(char) + s - 97) % 26 + 97) 
    return result

def decrypt(text, s):
    result = ""

    for i in range(len(text)):
        char = text[i]
        if (char.isupper()):
            result += chr((ord(char) - 65 - s) % 26 + 65)
        else:
            result += chr((ord(char) - 97 - s) % 26 + 97) 
    return result

choice = int(input("Encrypt (1) or Decrypt (2): "))
if (choice == 1):
    p = str(input("Enter your text: "))
    s = int(input("Enter the shift: "))
    cipher = encrypt(p, s)
    print("Plaintext: " + p)
    print("Ciphertext: "+ cipher)
else:
    p = str(input("Enter your text: "))
    s = int(input("Enter the shift: "))
    plain = decrypt(p, s)
    print("Ciphertext: " + p)
    print("Plaintext: "+ plain)
